data_steward/cdr_cleaner/cleaning_rules/fix_unmapped_survey_answers.py

```diff
@@ -0,0 +1,203 @@
+"""Original Issues: DC-1043, DC-1053 
+
+PPI answers are mapped to standard answer concepts in concept_relationship through 'Maps to value' (it has been this 
+case historically and it is still the case now), however, there are a bunch of PPI answer concepts (619 in total in 
+rdr20200807) missing such relationships in concept_relationship. Interestingly, the corresponding standard PPI 
+answer concepts could be found through 'Maps to'.  This might have been a bug in the vocabulary and they 
+probably should’ve used 'Maps to value' for mapping Answer concepts. 
+
+There are 619 unique answers (value_source_concept_ids) in observation that are mapped to a '0' value_as_concept_id, 
+out of which 368 are standard concepts, 202 are non-standard concepts that could be mapped to a standard concept 
+through Maps to in concept_relationship, and 49 are deprecated concepts that do not map to anything.
+
+For the 368 standard answers, we could just use it as-is for populating value_as_concept_id. However, among the 202 
```

Comment:
I might be getting thrown off by "however" on this line- to be sure, please let me know if this is correct. Here's a diagram to illustrate my understanding of how `value_as_concept_id` is derived for these rows. 
![image](https://user-images.githubusercontent.com/8049605/92633658-1689af80-f2a1-11ea-9e69-1c503e27024a.png)
We intend to replace the value in `value_as_concept_id` with that in `value_source_concept_id` whenever the latter is a standard concept. For rows whose `value_source_concept_id` refers to a non-standard concept, we populate `value_as_concept_id` with a standard concept obtained via `Maps to` (`maps_to_concept` in the diagram) which belongs to one of the appropriate classes (Context-dependent, Answer, Clinical Finding, Unit). No special treatment is needed for deprecated concepts.

Reply:
@mark-velez very nice diagram and this clearly illustrates the logic used in the query, this is way better than words (especially my words). 
This is a 100% accurate conceptual representation of the logic. I just wanted to mention that there might be a slight discrepancy in the query though, specifically for cases where `value_source_concept_id` is standard, the query replaces `value_as_concept_id` with the mapped concept from `concept_relationship`, which is the same as `value_source_concept_id` because the standard concepts map to themselves as well. This simplifies the query

Reply:
Thank you both, the diagram and description are super helpful in understanding this query!